dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/imports/TrackerImportControllerTest.java

```diff
@@ -165,6 +188,45 @@ public void verifySyncResponseShouldBeOkWhenImportReportStatusIsOk()
         }
     }
 
+    @Test
+    public void verifySyncResponseForCsvShouldBeOkWhenImportReportStatusIsOk()
+        throws Exception
+    {
+        // When
+        when( importStrategy.importReport( any() ) ).thenReturn( TrackerImportReportFinalizer.withImportCompleted(
+            TrackerStatus.OK,
+            TrackerBundleReport.builder()
+                .status( TrackerStatus.OK )
+                .build(),
+            TrackerValidationReport.builder()
+                .build(),
+            new TrackerTimingsStats(),
+            new HashMap<>() ) );
+
+        // Then
+        String contentAsString = mockMvc.perform( post( ENDPOINT + "?async=false&skipFirst=true" )
+            .content( "{}" )
+            .contentType( "text/csv" ) )
+            .andExpect( status().isOk() )
+            .andExpect( jsonPath( "$.message" ).doesNotExist() )
+            .andExpect( content().contentType( "application/json" ) )
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        verify( csvEventService ).readEvents( any(), eq( true ) );
+        verify( importStrategy ).importReport( any() );
+
+        try
+        {
+            renderService.fromJson( contentAsString, TrackerImportReport.class );
+        }
+        catch ( Exception e )
+        {
+            fail( "response content : " + contentAsString + "\n" + " is not of TrackerImportReport type" );
```

Comment:
Is this just to improve readability of the error message in case an Exception is thrown? Just asking since Junit would fail the test anyways if an exception is thrown but not expected

Reply:
It is just to give a cleaner message when the test fails